source/kraken/realtime.cpp

```diff
@@ -43,20 +43,58 @@ www.navitia.io
 namespace navitia {
 
 static bool is_handleable(const transit_realtime::TripUpdate& trip_update){
+    namespace bpt = boost::posix_time;
+
+    auto log = log4cplus::Logger::getInstance("realtime");
     // Case 1: the trip is cancelled
     if (trip_update.trip().schedule_relationship() == transit_realtime::TripDescriptor_ScheduleRelationship_CANCELED) {
         return true;
     }
     // Case 2: the trip is not cancelled, but modified (ex: the train's arrival is delayed or the train's
-    // departure is brought foward), we check the size of stop_time_update because we can't find a proper
+    // departure is brought forward), we check the size of stop_time_update because we can't find a proper
     // enum in gtfs-rt proto to express this idea
     if (trip_update.trip().schedule_relationship() == transit_realtime::TripDescriptor_ScheduleRelationship_SCHEDULED
             && trip_update.stop_time_update_size()) {
+        auto start_date = boost::gregorian::from_undelimited_string(trip_update.trip().start_date());
+        auto start_first_day_of_impact = bpt::ptime(start_date, bpt::time_duration(0, 0, 0, 0));
+        for (const auto& st: trip_update.stop_time_update()) {
+            auto ptime_arrival = bpt::from_time_t(st.arrival().time());
+            auto ptime_departure = bpt::from_time_t(st.departure().time());
+            if (ptime_arrival < start_first_day_of_impact
+                    || ptime_departure < start_first_day_of_impact) {
+                LOG4CPLUS_WARN(log, "Stop time " << st.stop_id() << " is before the day of impact");
+                return false;
+            }
+        }
         return true;
     }
     return false;
 }
 
+static bool check_trip_update(const transit_realtime::TripUpdate& trip_update) {
+    auto log = log4cplus::Logger::getInstance("realtime");
+    if (trip_update.trip().schedule_relationship() ==
+                                transit_realtime::TripDescriptor_ScheduleRelationship_SCHEDULED
```

Comment:
Yep just dealt with line length, looks bad, but, you know, open source code and all...